include/NAS2D/Delegate.h

```diff
@@ -236,19 +236,19 @@ class DelegateMemento
 
 public:
 #if !defined(FASTDELEGATE_USESTATICFUNCTIONHACK)
-	DelegateMemento() : m_pthis(0), m_pFunction(0), m_pStaticFunction(0) {};
-	void clear() { m_pthis = 0; m_pFunction = 0; m_pStaticFunction = 0; }
+	DelegateMemento() : m_pthis(nullptr), m_pFunction(nullptr), m_pStaticFunction(nullptr) {};
+	void clear() { m_pthis = nullptr; m_pFunction = nullptr; m_pStaticFunction = nullptr; }
 #else
-	DelegateMemento() : m_pthis(0), m_pFunction(0) {};
-	void clear() { m_pthis = 0; m_pFunction = 0; }
+	DelegateMemento() : m_pthis(nullptr), m_pFunction(nullptr) {};
+	void clear() { m_pthis = nullptr; m_pFunction = nullptr; }
 #endif
 public:
 #if !defined(FASTDELEGATE_USESTATICFUNCTIONHACK)
 	inline bool IsEqual(const DelegateMemento& x) const
 	{
 		if (m_pFunction != x.m_pFunction) return false;
 		if (m_pStaticFunction != x.m_pStaticFunction) return false;
-		if (m_pStaticFunction != 0) return m_pthis == x.m_pthis;
+		if (m_pStaticFunction != nullptr) return m_pthis == x.m_pthis;
```

Comment:
It's a common C++ idiom to compare pointers as conversions to bool, there's even a rule about it: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es87-dont-add-redundant--or--to-conditions . The second **Note** references using negation instead of `!=`

Reply:
I agree we shouldn't be overly verbose with such comparisons.

Maybe slightly out of scope for the change I intended, though it may be suitable to include here.